Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -107,22 +107,47 @@ def bump_formula_pr
       ]
     end
 
+    unless ARGV.dry_run?
+      backup_file = "#{formula.path}.orig"
+      if File.exist? backup_file
+        odie "Backup file #{backup_file} already exists!"
+      end
+      cp formula.path, "#{formula.path}.orig"
```

Comment:
I'd prefer if the backup resided in memory (`original_code = formula.path.read`). No need to check for file name collisions and no files that need to be cleaned up if anything goes wrong. And restoring is no more difficult than moving a file.